s2/s2_fuzz_test.go

```diff
@@ -0,0 +1,34 @@
+package s2
+
+import (
+	"github.com/klauspost/compress/internal/fuzz"
+	"testing"
+)
+
+func FuzzS2(f *testing.F) {
+	fuzz.AddFromZip(f, "testdata/fuzz/block-corpus-raw.zip", fuzz.TypeRaw, false)
+	f.Fuzz(func(t *testing.T, data []byte) {
+		concat, err := ConcatBlocks(nil, data, []byte{0})
+		if err != nil || concat == nil {
+			return
+		}
+
+		if len(data) < 9 { //to avoid errors when converting to uint64
+			return
+		}
+		EstimateBlockSize(data)
+		encoded := make([]byte, MaxEncodedLen(len(data)))
+		if len(encoded) < MaxEncodedLen(len(data)) || minNonLiteralBlockSize > len(data) || len(data) > maxBlockSize {
+			return
+		}
+
+		encodeBlockGo(encoded, data)
+		encodeBlockBetterGo(encoded, data)
+		encodeBlockSnappyGo(encoded, data)
+		encodeBlockBetterSnappyGo(encoded, data)
+		dst := encodeGo(encoded, data)
+		if dst == nil {
+			return
+		}
```

Comment:
So this "just" tests the encoders similar to `FuzzEncodingBlocks`, but avoids the assembly. I tend to just use tags for disabling assembly.

But if you want to add them, having them in that function makes sense to me.

Reply:
I'll move them to FuzzEncodingBlocks function later. Could you please clarify, what you mean by "but avoids the assembly" and "I tend to just use tags for disabling assembly."

I've included these encode* functions to cover them during fuzzing.